### PR TITLE
[E2E] Simplify CommentsAreaComponent and related specs

### LINKED ISSUE #51946: E2E: Flaky "Can post a reply" in wp-comments-spec.js
This spec seems to be flaky so let's try to fix it.

#### Failing spec details

From https://app.circleci.com/pipelines/github/Automattic/wp-e2e-tests-for-branches/64815/workflows/ce6a8563-bdf7-4853-8d29-dd6eb4fff2f1/jobs/155290:

```
[WPCOM] Comments: (mobile) Commenting and replying to newly created post in Gutenberg Editor: @parallel Can post a reply - Can post a reply
/home/circleci/wp-calypso/wp-calypso/test/e2e/specs-gutenberg/wp-comments-spec.js
ElementClickInterceptedError: element click intercepted: Element <form action="https://e2eflowtestinggutenbergsimple.wordpress.com/wp-comments-post.php" method="post" id="commentform" class="comment-form" novalidate="" style="background: gold; border: 2px solid red;">...</form> is not clickable at point (192, 967). Other element would receive the click: <div id="respond" class="comment-respond js" style="margin-left: 0px; margin-right: 0px;">...</div>
  (Session info: chrome=84.0.4147.135)
    at Object.throwDecodedError (/home/circleci/wp-calypso/wp-calypso/node_modules/selenium-webdriver/lib/error.js:550:15)
    at parseHttpResponse (/home/circleci/wp-calypso/wp-calypso/node_modules/selenium-webdriver/lib/http.js:565:13)
    at Executor.execute (/home/circleci/wp-calypso/wp-calypso/node_modules/selenium-webdriver/lib/http.js:491:26)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Driver.execute (/home/circleci/wp-calypso/wp-calypso/node_modules/selenium-webdriver/lib/webdriver.js:700:17)
    at async Proxy.clickWhenClickable (lib/driver-helper.js:73:3)
    at async CommentsAreaComponent._postComment (lib/pages/frontend/comments-area-component.js:26:3)
    at async CommentsAreaComponent.reply (lib/pages/frontend/comments-area-component.js:45:3)
    at async Context.<anonymous> (specs-gutenberg/wp-comments-spec.js:65:4)
```

### PR DESCRIPTION
**Pull request by fullofcaffeine**
#### Changes proposed in this Pull Request

Started as an investigation on how to fix https://github.com/Automattic/wp-calypso/issues/51946.

I could see why the spec would fail, but I couldn't reproduce the exact same error locally. I then just tried to investigate ways to tidy it up. 

Document the methods in `CommentsAreaComponent`, too.

#### Testing instructions

* The specs should pass :)

Possibly fixes https://github.com/Automattic/wp-calypso/issues/51946.

### CODE REVIEW

File: test/e2e/lib/pages/frontend/comments-area-component.js [hunk 1]
```diff
@@ -16,50 +15,46 @@ export default class CommentsAreaComponent extends AsyncBaseContainer {
...
+	 * Verifies if a comment is visible. If you tweak the depth, you can also
+	 * "point" it to a reply comment.
+	 *
+	 * @param { string } comment The comment text.
+	 * @param { number } depth How nested you want to check. The first comment has
+	 * a depth of 1, while the first reply to it would have a depth of 2.
+	 */
+	async verifyCommentIsVisible( comment, depth = 1 ) {
+		const commentLocator = By.xpath(
+			`//li[contains(@class,'depth-${ depth }')]//div[@class='comment-content']/p[.='${ comment }']`
```
Comment (fullofcaffeine) [inline test/e2e/lib/pages/frontend/comments-area-component.js:55]: There wasn't a need for two slightly similar xpath locators in two different functions, so I extracted the logic into this helper. Also, I thought it'd best not to couple the responsibility of asserting the result with the side-effect functions (`_postComment` and `reply`). Each test now has the responsibility of asserting the result by calling this function, see [this line](https://github.com/Automattic/wp-calypso/pull/52217/files#diff-0b7e3b134038ca1288f2be96f6db46367a456909d110cbc9c2b9c7b80785b522R60) as an example.

File: test/e2e/lib/pages/frontend/comments-area-component.js [hunk 2]
```diff
@@ -16,50 +15,46 @@ export default class CommentsAreaComponent extends AsyncBaseContainer {
...
 	}
 
-	async _postComment( { comment } ) {
-		const commentForm = By.css( '#commentform' );
+	/**
+	 * Posts the first comment to a post (under "Join the Conversation" in the footer).
+	 *
+	 * @param { string } comment The comment text.
+	 */
+	async _postComment( comment ) {
```
Comment (fullofcaffeine) [inline test/e2e/lib/pages/frontend/comments-area-component.js:23]: Wondering if it'd make sense to rename this function to `postComment`. Does anyone know the reason for the `_` prefix?

File: test/e2e/specs-gutenberg/wp-comments-spec.js [hunk 3]
```diff
@@ -53,23 +53,20 @@ describe( `[${ host }] Comments: (${ screenSize })`, function () {
 
 		step( 'Can post a comment', async function () {
 			const commentArea = await CommentsAreaComponent.Expect( driver );
-			return await commentArea._postComment( {
-				comment: dataHelper.randomPhrase(),
-				name: 'e2eTestName',
-				email: 'e2eTestName@test.com',
```
Comment (fullofcaffeine) [inline test/e2e/specs-gutenberg/wp-comments-spec.js:59]: The `name` and `email` attributes didn't seem to be used at all. In fact, the `_postComment` function was de-structuring the object into its comment attribute, see [#](https://github.com/Automattic/wp-calypso/pull/52217/files#diff-3871ba5bc6b03a90dfdba78b63a47b287941a37d295277cd7b2327d582506bafL19).

File: test/e2e/lib/pages/frontend/comments-area-component.js [hunk 4]
```diff
@@ -16,50 +15,46 @@ export default class CommentsAreaComponent extends AsyncBaseContainer {
...
 		const commentField = By.css( '#comment' );
 		const submitButton = By.css( '.form-submit #comment-submit' );
-		const commentContent = By.xpath( `//div[@class='comment-content']/p[.='${ comment }']` );
 
-		await driverHelper.scrollIntoView( this.driver, commentForm, 'end' );
-		await driverHelper.clickWhenClickable( this.driver, commentForm );
-		await driverHelper.scrollIntoView( this.driver, commentForm, 'end' );
-		await this.switchToFrameIfJetpack();
```
Comment (fullofcaffeine) [inline test/e2e/lib/pages/frontend/comments-area-component.js:28]: I've checked with @brbrr about this, see: p1619052802005100-slack-C1A1EKDGQ. Looks like it's not used anymore.

**Comment by WunderBart** [comment]
> I could see why the spec would fail, but I couldn't reproduce the exact same error locally. I then just tried to investigate ways to tidy it up.

From the original error (#51946):

```
ElementClickInterceptedError: element click intercepted: Element <form action="https://e2eflowtestinggutenbergsimple.wordpress.com/wp-comments-post.php" method="post" id="commentform" class="comment-form" novalidate="" style="background: gold; border: 2px solid red;">...</form> is not clickable at point (192, 967). Other element would receive the click: <div id="respond" class="comment-respond js" style="margin-left: 0px; margin-right: 0px;">...</div>
```

An intercepted element usually means that the target element has been covered by some other element or that it's still moving when clicked and it's missed. Do you know what could be the cause in this case?

File: test/e2e/lib/pages/frontend/comments-area-component.js (hunk 2 above)
Comment (WunderBart) [reply test/e2e/lib/pages/frontend/comments-area-component.js:23]: Probably to view it as a private method? 🤷‍♂️ Anyway, I'm for removing the underscore prefix 👍

**Comment by fullofcaffeine** [comment]
@WunderBart Thanks a lot for taking the time to review 🙇🏻 

>An intercepted element usually means that the target element has been covered by some other element or that it's still moving when clicked and it's missed. Do you know what could be the cause in this case?

I don't. The code was quite convoluted before as you might see in the diff. I did try to reproduce and pinpoint what could be getting on the way but couldn't. There's a good chance that the cause was in some of the code I removed, though. I'd say let's merge this one and keep watching :)

**Review by WunderBart — APPROVED** [review]
> I don't. The code was quite convoluted before as you might see in the diff. I did try to reproduce and pinpoint what could be getting on the way but couldn't. There's a good chance that the cause was in some of the code I removed, though. I'd say let's merge this one and keep watching :)

Sounds good! 🚢

**Pull request merged**